overlord/state/task.go

```diff
@@ -143,17 +143,15 @@ func (t *Task) State() *State {
 }
 
 // Progress returns the current progress for the task.
-// If progress is not explicitly set, it returns (0, 1) if the status is
-// RunningStatus or WaitingStatus and (1, 1) otherwise.
+// If progress is not explicitly set, it returns
+// (0, 1) if the status is DoStatus and (1, 1) otherwise.
 func (t *Task) Progress() (cur, total int) {
 	t.state.ensureLocked()
 	if t.progress == nil {
-		switch t.Status() {
-		case RunningStatus, WaitingStatus:
+		if t.Status() == DoStatus {
 			return 0, 1
```

Comment:
perhaps on undone status we could return (0, 1) to indicate that tasks have been "undone"